bee-network/src/swarm/protocols/iota_gossip/id.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+// use std::convert::TryFrom;
+
+use libp2p_core::ProtocolName;
+
+use std::fmt;
+
+#[derive(Debug, Clone)]
+pub struct IotaGossipIdentifier {
+    name: String,
+    network_id: u64,
+    version: String,
+    buffered: Vec<u8>,
+}
+
+impl IotaGossipIdentifier {
+    pub fn new(name: String, network_id: u64, version: String) -> Self {
+        let buffered = format!("/{}/{}/{}", name, network_id, version).into_bytes();
+
+        Self {
+            name,
+            network_id,
+            version,
+            buffered,
+        }
+    }
+}
+
+impl ProtocolName for IotaGossipIdentifier {
+    fn protocol_name(&self) -> &[u8] {
+        &self.buffered
+    }
+}
+
+impl fmt::Display for IotaGossipIdentifier {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // Panic:
+        // Unwrapping is fine, because we made sure `buffered` contains a valid Utf8 string.
+        write!(f, "{}", String::from_utf8(self.buffered.clone()).unwrap())
```

Comment:
Maybe its not worth the trouble. But you could store `buffered` as a `String` and just use `as_bytes` for `protocol_name`

Reply:
nice...I didn't like it either the next day, and wanted to do just as you propose here.

Reply:
Done. The type could just be turned into a wrapper around a String. I wasn't sure initially whether I would need the individual parts as well of the composed string as well, but it turned out to not be the case :smiley: